core/domain/state_domain_test.py

```diff
@@ -544,3 +562,149 @@ def test_solution_validation(self):
+                invalid_content_id)
+
+    def test_delete_content_id_for_translations_deletes_content_id(self):
+        old_content_translations_dict = {
+            'content': {
+                'en': {
+                    'html': 'hello!',
+                    'needs_update': False
+                }
+            }
+        }
+
+        content_translations = state_domain.ContentTranslations.from_dict(
+            old_content_translations_dict)
+        content_translations.delete_content_id_for_translation('content')
+
+        self.assertEqual(len(content_translations.to_dict()), 0)
+
+    def test_delete_content_id_for_translation_with_invalid_content_id_raise_error(self): # pylint: disable=line-too-long
+        content_translations = state_domain.ContentTranslations.from_dict({})
+        invalid_content_id_to_delete = 123
+        with self.assertRaisesRegexp(
+            Exception, 'Expected content_id to be a string, '):
+            content_translations.delete_content_id_for_translation(
+                invalid_content_id_to_delete)
+
+    def test_validation_with_invalid_content_id_raise_error(self):
+        content_translations_dict = {
+            123: {}
+        }
+
+        content_translations = state_domain.ContentTranslations.from_dict(
+            content_translations_dict)
+
+        self._assert_validation_error(
+            content_translations, 'Expected content_id to be a string, ')
+
+    def test_validation_with_invalid_type_langauge_code_raise_error(self):
+        content_translations_dict = {
+            'content': {
+                123: {
+                    'html': 'hello!',
+                    'needs_update': False
+                }
+            }
+        }
+
+        content_translations = state_domain.ContentTranslations.from_dict(
+            content_translations_dict)
+
+        self._assert_validation_error(
+            content_translations, 'Expected language_code to be a string, ')
+
+    def test_validation_with_unknown_langauge_code_raise_error(self):
+        content_translations_dict = {
+            'content': {
+                'ed': {
+                    'html': 'hello!',
+                    'needs_update': False
+                }
+            }
+        }
+
+        content_translations = state_domain.ContentTranslations.from_dict(
+            content_translations_dict)
+
+        self._assert_validation_error(
+            content_translations, 'Invalid language_code: ed')
+
+    def test_validation_with_3_letter_langauge_code_raise_error(self):
+        content_translations_dict = {
+            'content': {
+                'xyz': {
+                    'html': 'hello!',
+                    'needs_update': False
+                }
+            }
+        }
+
+        content_translations = state_domain.ContentTranslations.from_dict(
+            content_translations_dict)
+
+        # Adding a 3-letter language code in ALL_LANGUAGE_CODES for validation.
```

Comment:
Try not to do this. Use `with self.swap()` instead.

Reply:
Removed this test entirely as we can accept 3-letter language code. (If present in ALL_LANGUAGE_CODES)